### PR TITLE
Server config file

### PR DESCRIPTION
**Pull request by jenr24G**
Creates functionality to allow the server to use a `yaml` config file in addition to the command line args, for more ergonomic use of the Server, as well as to support the upcoming Regression/UAT suite
* Add `Deserialize` to `ServerArgs` and create a `deserialize_level_filter` helper function
* Extend the Server `clap` parser to use two new subcommands: `mina-indexer server cli`, which works the same as `mina-indexer server` did before, and `config` which just takes a path to a `yaml` config file: `mina-indexer server config --path ./example.config.yaml`

<hr>

Asana Tasks referenced:
* https://app.asana.com/0/1203669416123378/1204910036761031/f
* https://app.asana.com/0/1203669416123378/1205188527548527/f

### CODE REVIEW

File: example.config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+root_hash: 3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ
+startup_dir: /home/jenr/.mina-indexer/startup-blocks
+watch_dir: /home/jenr/.mina-indexer/watch-blocks
+database_dir: /home/jenr/.mina-indexer/database
+log_dir: /home/jenr/.mina-indexer/logs
+keep_non_canonical_blocks: true
+log_level: info
+log_level_stdout: trace
+prune_interval: 10
+canonical_update_threshold: 2
```
Comment (trevorbernard) [inline example.config.yaml:12]: Can you use env variables? s/jenr/$HOME/g

Comment (jenr24G) [reply example.config.yaml:12]: No, the paths are parsed directly into `PathBuf`s. We'd need to write functionality to iterate through a `PathBuf` and replace `$ENV` patterns with their values using `std::env::var`. Cool feature, but it shouldn't block the merge of this PR

https://app.asana.com/0/1203669416123378/1205272524183351/f

Comment (jenr24G) [reply example.config.yaml:12]: also what in the world is s//g

Comment (trevorbernard) [reply example.config.yaml:12]: global search and replace

Comment (jenr24G) [reply example.config.yaml:12]: @trevorbernard

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**